tools/toktx/toktx.cc

```diff
@@ -724,7 +724,7 @@ toktxApp::main(int argc, _TCHAR *argv[])
             image =
               Image::CreateFromFile(infile,
                                     options.assign_oetf == KHR_DF_TRANSFER_UNSPECIFIED,
-                                    options.bcmp || options.bopts.uastc);
+                                    options.bcmp || options.bopts.uastc || options.astc);
```

Comment:
Fixed in https://github.com/KhronosGroup/KTX-Software/pull/433/commits/2bf7bc64f0c47c160ec22c32923f30061bd6599d

Reply:
I think it is better to default to HDR if the input is 16 bit. If someone is explicitly specifying LDR with 16-bit input a warning is not necessary.

Some questions:

1. If the encoder gets 8-bit input and HDR is set, what happens? Wouldn't it make LDR blocks regardless of the HDR setting?
2. Is there ever a reason to rescale 8-bit input to 16 and specify HDR? I don't think there is but want to confirm.

Assuming the answer to no. 2 is no  then the only rescaling we need to do is from < 8-bits to 8 bits.

Reply:
Partially fixed in https://github.com/KhronosGroup/KTX-Software/pull/433/commits/18af4983456764eb784babe2d11a125f5d2b7f5c. 

Answering the questions;

1. I am not fully sure but If HDR mode is set it will create HDR blocks from the 8bit textures provided. This is more of an output format option than input format option.
2. Technically no. I mean you can't create data for missing bits. But in HDR mode the astc calculations might be in higher precision internally. 

I am not fully sure about either of the answer so still trying to find out but these answers come from my reasoning really.

Reply:
1. I'm only familiar with ASTC behaviour from the consuming end and I know that LDR/HDR is on a block-by-block basis so there is no need for the encoder to encode everything to HDR blocks.
2. I don't think the calculations will be beneficial to 8-bit input.

Reply:
Answers to your both original questions is "No" according to Peter Harris, the maintainer of astcenc.

Reply:
> Also replied to the other comment. I think we should close one of these. Perhaps this one and keep the other one for discussion.

Yes. I've responded there.